operator/pkg/helmreconciler/prune.go

```diff
@@ -43,6 +43,8 @@ import (
 var (
 	// NamespacedResources orders non cluster scope resources types which should be deleted, first to last
 	NamespacedResources = []schema.GroupVersionKind{
+		{Group: "autoscaling", Version: "v2", Kind: name.HPAStr},
```

Comment:
instead of 2, can we have a single one and make NamespacedResources not a constant but rather function of k8s version?

As is we get that confusing warning and I think 2x the amount of API server calls.

Reply:
done